### PR TITLE
15.0 mig mgmtsystem audit

### PR DESCRIPTION
**Pull request by YvanDotet**

### CODE REVIEW

File: mgmtsystem_audit/data/audit_automated_actions.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<odoo noupdate="1">
+    <!-- Email template for "Prepare to audit" message to all auditors. -->
+    <record id="template_email_prepare_to_audit" model="mail.template">
+        <field name="name">Prepare to audit</field>
+        <field name="model_id" ref="model_mgmtsystem_audit" />
+        <field
+            name="email_to"
+        >{{",".join(object.auditor_user_ids.mapped("email"))}}</field>
+        <!--<field name="email_to">{{('"%s" &lt;%s&gt;' % (object.partner_id.name or object.name, object.partner_id.email or object.email)) | safe}}</field>-->
```
Comment (ivantodorovich) [inline mgmtsystem_audit/data/audit_automated_actions.xml]:
```suggestion
```

Doesn't seem useful, so let's remove it IMO

File: mgmtsystem_audit/models/mgmtsystem_audit.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+from odoo import _, api, fields, models
+
+
+class MgmtsystemAudit(models.Model):
+    """Model class that manage audit."""
+
+    _name = "mgmtsystem.audit"
+    _description = "Audit"
+    _inherit = ["mail.thread"]
```
Comment (ivantodorovich) [inline mgmtsystem_audit/models/mgmtsystem_audit.py]: nitpicking: I'd take the opportunity to add the activity mixing here, too
What do you think?

File: mgmtsystem_audit/models/mgmtsystem_audit.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+    name = fields.Char()
+    reference = fields.Char(size=64, required=True, readonly=True, default="NEW")
+    date = fields.Datetime()
+    line_ids = fields.One2many(
+        "mgmtsystem.verification.line", "audit_id", "Verification List"
+    )
+    number_of_audits = fields.Integer("# of audits", readonly=True, default=1)
+    number_of_nonconformities = fields.Integer(
+        readonly=True,
```
Comment (ivantodorovich) [inline mgmtsystem_audit/models/mgmtsystem_audit.py]:
```suggestion
```
Computed fields are readonly by default

File: mgmtsystem_audit/models/mgmtsystem_audit.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        store=True,
+        compute="_compute_number_of_nonconformities",
+    )
+    number_of_questions_in_verification_list = fields.Integer(
+        readonly=True,
```
Comment (ivantodorovich) [inline mgmtsystem_audit/models/mgmtsystem_audit.py]:
```suggestion
```
Computed fields are readonly by default

File: mgmtsystem_audit/models/mgmtsystem_audit.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        store=True,
+        compute="_compute_number_of_questions_in_verification_list",
+    )
+    number_of_improvements_opportunity = fields.Integer(
+        "Number of improvements Opportunities",
+        readonly=True,
```
Comment (ivantodorovich) [inline mgmtsystem_audit/models/mgmtsystem_audit.py]:
```suggestion
```
Computed fields are readonly by default

File: mgmtsystem_audit/models/mgmtsystem_audit.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        store=True,
+        compute="_compute_number_of_improvement_opportunities",
+    )
+    days_since_last_update = fields.Integer(
+        readonly=True,
```
Comment (ivantodorovich) [inline mgmtsystem_audit/models/mgmtsystem_audit.py]:
```suggestion
```
Computed fields are readonly by default

File: mgmtsystem_audit/models/mgmtsystem_audit.py [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+        store=True,
+        compute="_compute_days_since_last_update",
+    )
+    closing_date = fields.Datetime(readonly=True)
+
+    number_of_days_to_close = fields.Integer(
+        "# of days to close",
+        readonly=True,
```
Comment (ivantodorovich) [inline mgmtsystem_audit/models/mgmtsystem_audit.py]:
```suggestion
```
Computed fields are readonly by default

**Review by ivantodorovich — APPROVED** [review]

File: mgmtsystem_audit/models/mgmtsystem_audit.py [hunk 8]
```diff
@@ -82,9 +77,7 @@ class MgmtsystemAudit(models.Model):
     nonconformity_ids = fields.Many2many(
         "mgmtsystem.nonconformity", string="Nonconformities"
     )
-    state = fields.Selection(
-        [("open", "Open"), ("done", "Closed")], "State", default="open"
-    )
+    state = fields.Selection([("open", "Open"), ("done", "Closed")], default="open")
```
Comment (ivantodorovich) [inline mgmtsystem_audit/models/mgmtsystem_audit.py]: This field should probably be required

File: mgmtsystem_audit/models/mgmtsystem_audit.py (hunk 2 above)
Comment (YvanDotet) [reply mgmtsystem_audit/models/mgmtsystem_audit.py]: It was not in mgmtsystem hazard but was in mgmtsystem action.
We can add it because it is audit for audition and it is important.

**Review by ivantodorovich — APPROVED** [review]

**Comment by YvanDotet** [comment]
Hello,
I think this module is ready for merging, isn't it ?
Thank you.

**Comment by YvanDotet** [comment]
Hello,
I think it's ok to merge,
Thank you.

**Comment by ivantodorovich** [comment]
Hello! It is, but we just need one more review

**Comment by YvanDotet** [comment]
Hello,
Ok, thank you to you.

Sincerely,
Yvan.

File: mgmtsystem_audit/__manifest__.py [hunk 9]
```diff
@@ -22,5 +22,5 @@
         "wizard/copy_verification_lines.xml",
     ],
     "demo": ["demo/demo_audit.xml"],
-    "development_status": "Production/Stable",
```
Comment (etobella) [inline mgmtsystem_audit/__manifest__.py]: Why did you remove this?

File: mgmtsystem_audit/__manifest__.py [hunk 10]
```diff
@@ -22,5 +22,5 @@
...
+    "installable": True,
```
Comment (etobella) [inline mgmtsystem_audit/__manifest__.py]: This is not necessary

**Review by etobella — APPROVED** [review]
Just two small comments. It LGTM in general (I approve it expecting the comments :smile: )

File: mgmtsystem_audit/__manifest__.py (hunk 10 above)
Comment (YvanDotet) [reply mgmtsystem_audit/__manifest__.py]: ok I change this

File: mgmtsystem_audit/__manifest__.py (hunk 9 above)
Comment (YvanDotet) [reply mgmtsystem_audit/__manifest__.py]: When I remove this, the message become : 
"mgmtsystem_audit (Production/Stable) depends on mgmtsystem_nonconformity (Beta)"

That's why I had removed it.

File: mgmtsystem_audit/__manifest__.py (hunk 10 above)
Comment (ivantodorovich) [reply mgmtsystem_audit/__manifest__.py]: Aha! so this is why you removed it.
The CI bot that checks for dev status inconsistencies is complaining:

```
Run manifestoo -d . check-dev-status --default-dev-status=Beta
mgmtsystem_audit (Production/Stable) depends on mgmtsystem_nonconformity (Beta)
```

Basically, we can't have Production/Stable modules depend on Beta modules.

I think we should bump `mgmtsystem_nonconformity` to Production/Stable instead, though. It's time.

Comment (YvanDotet) [reply mgmtsystem_audit/__manifest__.py]: Hello @ivantodorovich 
Ok thanks what should I do ?

Comment (ivantodorovich) [reply mgmtsystem_audit/__manifest__.py]: Actually, on a second thought.. 

The dependency chain is a nightmare, and none of the depended modules have the `development_status` set on manifest. For this module to be Production/Stable, we need all other modules in the dependency chain to be Production/Stable as well. And these aren't just modules on this repo, also on knowledge repo (`document_page`)

I vote you simply go back to removing the `development_status` key on this PR. We can see later about bumping all of them, starting from the ones in knowledge repository

Comment (YvanDotet) [reply mgmtsystem_audit/__manifest__.py]: @ivantodorovich 
Ok I go back to the previous development and it show green now.
Thank.

**Comment by ivantodorovich** [comment]
Thanks!

/ocabot merge nobump
